File: output/gemma4-e2b_IO_1/avatar/Python/Java/codeforces_387_A.java
import java.util.Scanner;

public class codeforces_387_A {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        // Read first line (h1:m1)
        String line1 = scanner.nextLine();
        String[] parts1 = line1.split(":");
        int h1 = Integer.parseInt(parts1[0]);
        int m1 = Integer.parseInt(parts1[1]);

        // Read second line (h2:m2)
        String line2 = scanner.nextLine();
        String[] parts2 = line2.split(":");
        int h2 = Integer.parseInt(parts2[0]);
        int m2 = Integer.parseInt(parts2[1]);

        // Calculate total minutes for time 1
        int totalM1 = m1 + h1 * 60;

        // Calculate total minutes for time 2
        int totalM2 = m2 + h2 * 60;

        // Calculate the difference in minutes
        int diffMinutes = totalM1 - totalM2;

        // Apply modulo 1440 (minutes in a day)
        int finalMinutes = diffMinutes % 1440;

        // Convert the difference back to hours and minutes
        int finalHours = finalMinutes / 60;
        int finalMinutesPart = finalMinutes % 60;

        // Print the result in "%02d:%02d" format
        System.out.printf("%02d:%02d%n", finalHours, finalMinutesPart);
    }
}

File: output/gemma4-e2b_IO_1/avatar/Java/Python/codeforces_387_A.py
import sys

def solve():
    # Read input from stdin
    try:
        line1 = sys.stdin.readline().strip()
        line2 = sys.stdin.readline().strip()
    except EOFError:
        return

    if not line1 or not line2:
        return

    # Parse line 1 (x)
    try:
        h1_str, m1_str = line1.split(':')
        h1 = int(h1_str)
        m1 = int(m1_str)
    except ValueError:
        # Handle potential parsing errors if input format is strictly guaranteed, this might not be necessary
        return

    # Parse line 2 (y)
    try:
        h2_str, m2_str = line2.split(':')
        h2 = int(h2_str)
        m2 = int(m2_str)
    except ValueError:
        return

    # Calculate the result based on the Java logic: x.minusHours(y.getHour()).minusMinutes(y.getMinute())
    # x = h1:m1, y = h2:m2
    
    # Calculate the difference in total minutes
    # x in minutes: h1 * 60 + m1
    # y in minutes: h2 * 60 + m2
    
    x_minutes = h1 * 60 + m1
    y_minutes = h2 * 60 + m2
    
    # Calculate the difference in minutes
    diff_minutes = x_minutes - y_minutes
    
    # Convert the difference back to hours and minutes
    # Since the Java code uses LocalTime subtraction, we need to handle potential negative results 
    # if the subtraction crosses midnight, although for this specific problem structure, 
    # the result is expected to be positive and within the 24-hour cycle.
    
    # Total minutes in a day
    MINUTES_PER_DAY = 24 * 60
    
    # Ensure the result is positive and within the 24-hour cycle (modulo arithmetic)
    # Python's % operator handles negative numbers correctly for mathematical modulo if the divisor is positive.
    result_minutes = diff_minutes % MINUTES_PER_DAY
    
    result_h = result_minutes // 60
    result_m = result_minutes
